tangelo/algorithms/classical/mp2_solver.py

```diff
@@ -132,12 +147,184 @@ def get_mp2_amplitudes(self):
+        self.mp2wfn = None
+
+        self.molecule = SecondQuantizedMolecule(xyz=molecule.xyz, q=molecule.q, spin=molecule.spin,
+                                                solver=IntegralSolverPsi4(),
+                                                basis=molecule.basis,
+                                                ecp=molecule.ecp,
+                                                symmetry=False,
+                                                uhf=molecule.uhf,
+                                                frozen_orbitals=molecule.frozen_orbitals)
+        self.basis = molecule.basis
+
+    def simulate(self):
+        """Perform the simulation (energy calculation) for the molecule.
+
+        Returns:
+            float: Total MP2 energy.
+        """
+        n_frozen_vir = len(self.molecule.frozen_virtual)
+        n_frozen_occ = len(self.molecule.frozen_occupied)
+        if n_frozen_occ or n_frozen_vir:
+            if self.molecule.uhf:
+                if (set(self.molecule.frozen_occupied[0]) != set(self.molecule.frozen_occupied[1]) or
+                   set(self.molecule.frozen_virtual[0]) != set(self.molecule.frozen_virtual)):
+                    raise ValueError(f"Only identical frozen orbitals for alpha and beta are supported in {self.__class__.__name__}")
+            focc = np.array(self.molecule.frozen_occupied)
+            fvir = np.array(self.molecule.frozen_virtual)
+            if np.any(focc > n_frozen_occ-1) or np.any(fvir < self.molecule.n_mos-n_frozen_vir):
+                raise ValueError(f"{self.__class__.__name__} does not support freezing interior orbitals")
+
+        if not self.molecule.uhf:
+            ref = 'rhf' if self.molecule.spin == 0 else 'rohf'
+        else:
+            ref = 'uhf'
+        self.backend.set_options({'basis': self.basis, 'mcscf_maxiter': 300, 'mcscf_diis_start': 20,
+                                  'opdm': True, 'tpdm': True, 'frozen_docc': [n_frozen_occ], 'frozen_uocc': [n_frozen_vir],
+                                  'reference': ref})
+
+        energy, self.mp2wfn = self.backend.energy('mp2', molecule=self.molecule.solver.mol,
+                                                  basis=self.basis, return_wfn=True)
+        return energy
+
+    def get_rdm(self):
+        """Calculate the 1- and 2-particle reduced density matrices.
+
+        Obtaining MP2 rdms from Psi4 is not supported in Tangelo.
+        
+        Raises:
+            RuntimeError: Not implemented at this time"""
+        raise RuntimeError("Returning MP2 rdms from Psi4 is not currently supported in Tangelo")
+
+    def get_mp2_amplitudes(self):
+        """Compute the double amplitudes from the MP2 perturbative method, and
+        then reorder the elements into a dense list. The single (T1) amplitudes
+        are set to a small non-zero value. The ordering is single, double
+        (diagonal), double (non-diagonal).
+
+        Returning MP2 amplitudes from Psi4 is not supported in Tangelo
```

Comment:
If someone tries to call UCCSD, it will call `get_amplitudes` by default and then it will return an error that `MP2SolverPsi4` does not have a `get_amplitudes` method. This should be more informative. 

If we want, it should be possible to hand code both RDM and amplitudes for RHF closed shell systems using         Using https://github.com/psi4/psi4numpy/blob/master/Tutorials/10_Orbital_Optimized_Methods/10a_orbital-optimized-mp2.ipynb. I have added this link in the docstring.